### PR TITLE
Update aws-sdk-go

### PR DESCRIPTION
**Pull request by patstrom**
**What this PR does / why we need it**:

With the recent release of [IAM roles for service accounts](https://github.com/aws/containers-roadmap/issues/23) the AWS sdk needs to be updated to at least [1.23.13](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts-minimum-sdk.html) in order to utilize this feature. 

**Which issue(s) this PR fixes**:

None.

**Special notes for your reviewer**:

### CODE REVIEW

**Review by marefr — CHANGES_REQUESTED** [review]
Could you please explain a bit more what this feature means and how it will be used.

You need to vendor the dependencies as well, see https://github.com/grafana/grafana/blob/master/UPGRADING_DEPENDENCIES.md#go-dependencies.

**Comment by patstrom** [comment]
I've updated the vendor, sorry!

We currently run everything in Kubernetes with and Amazon hosted control plane through EKS. In order to link an in-cluster pod to a IAM role we have been using kube2iam which requires a deamonset. Just recently AWS announced their own, native solution to linking IAM credentials to in-cluster resources which we are currently migrating to.

In short their solution is to link a Kubernetes service account to an IAM role using the TokenRequestProjection beta feature to automatically mount an OIDC token to the pods running as this service account. The `AWS_IAM_ROLE_ARN` and `AWS_WEB_IDENTITY_TOKEN_FILE` environment variables will be injected (and the token file automatically mounted) so that the pod can authenticate using `sts:AssumeRoleWithWebIdentity`. However, reading these environment variables and authenticating with web identity is only part of the default credentials chain since `aws-sdk-go@v1.23.13`.

Basically, we are moving from kube2iam to the AWS solution and for that everything in our cluster needs to run at least `aws-sdk-go@v1.23.13` (or equivalent in other languages).

**Comment by marefr** [comment]
@patstrom thanks for the additional information. Makes sense. You're sure that there are no other changes needed for the CloudWatch datasource for this to work? I guess what I'm asking is if you have tried these changes in container on AWS?

There's a linting error reported which fails the build. Probably due to updating of ini depenendency. Please fix for us to be able to review this again.

```
lint via golangci-lint
pkg/setting/setting.go:372:24: SA1019: ini.DEFAULT_SECTION is deprecated: Use "DefaultSection" instead.  (staticcheck)
		if section.Name() == ini.DEFAULT_SECTION {
```

**Comment by patstrom** [comment]
I fixed the linter error!

To be honest I haven't really confirmed with Grafana specifically but I've gone through all of our production applications doing essentially the same patch and there are similar merge requests for other projects (which I have confirmed are working in our production environment). E.g [external-dns](https://github.com/kubernetes-incubator/external-dns/pull/1182) and [alb-ingress-controller](https://github.com/kubernetes-sigs/aws-alb-ingress-controller/pull/1026).

Of course for this to work you would have to enable OIDC provider for you cluster and create service accounts with the correct annotation and IAM roles with the correct trust policy as described in [the AWS documentation](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html)

**Comment by patstrom** [comment]
@marefr What is the status on this?

**Comment by marefr** [comment]
Sorry. We'll review this eventually. Looking at the dependencies updated like /protobuf v1.3.2, Not sure that breaks something else. How did you/what commands used to upgrade the AWS dependency?

**Comment by patstrom** [comment]
I followed the commands in the documentation you linked.

**Comment by marefr** [comment]
Hmm okay. We have another PR updating dependencies that we want to merge to master. That would happen next week and would probably require the changes in this PR to be rebased on master and updating the aws-sdk dependency one more time. I'll let you know when the other PR is merged.

**Comment by patstrom** [comment]
Just include this update in that PR then?

**Comment by marefr** [comment]
If you merge in latest/rebase on master and update aws sdk again we can review/test it out to verify it doesn't break something. Thanks

**Comment by patstrom** [comment]
I rebased this on master and everything seems to be fine!

**Review by marefr — APPROVED** [review]
LGTM

**Comment by marefr** [comment]
Thank you for contributing to Grafana!
